### PR TITLE
Parameterise Roslyn and NuGet package versions

### PR DESCRIPTION
**Pull request by duncanp-sonar**

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-  <!-- Common settings for test and product code-->
+
+    <!-- Common settings for test and product code-->
 
   <PropertyGroup>
     <Version>3.0.0</Version>
     <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
+
+    <RoslynPackagesVersion>4.0.1</RoslynPackagesVersion>
+    <NuGetPackagesVersion>6.3.1</NuGetPackagesVersion>
```
Comment (duncanp-sonar) [inline Directory.Build.props:10]: Miscellaneous: defines the packages to use in a single place to make them easier to update.

File: Tests/RoslynPluginGeneratorTests/SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests.csproj [hunk 2]
```diff
@@ -16,7 +16,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="NuGet.CommandLine" Version="6.3.1" GeneratePathProperty="true" />
+    <PackageReference Include="NuGet.CommandLine" Version="$(NuGetPackagesVersion)" GeneratePathProperty="true" />
```
Comment (pavel-mikula-sonarsource) [inline Tests/RoslynPluginGeneratorTests/SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests.csproj:19]: Does this need to be a separate ItemGroup? I'd merge them like the other ones

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

File: Tests/RoslynPluginGeneratorTests/SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests.csproj (hunk 2 above)
Comment (duncanp-sonar) [reply Tests/RoslynPluginGeneratorTests/SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests.csproj:19]: Personally, I'd prefer to separate the other ones out  so they're like this one.
But I don't think it's important enough to do now.

**Pull request merged**